### ISSUE TITLE
An impressive article

### ISSUE THREAD

**Issue by CeciAc** [dotnet/docs#12157]
@dimiste commented on [Wed May 01 2019](https://github.com/dotnet/docs.es-es/issues/128)

My question is: in the piece of code:

```csharp
static async Task Main(string[] args)
{
    Coffee cup = PourCoffee();
    Console.WriteLine("coffee is ready");
    var eggsTask = FryEggsAsync(2);
    var baconTask = FryBaconAsync(3);
    var toastTask = makeToastWithButterAndJamAsync(2);

    var eggs = await eggsTask;
    Console.WriteLine("eggs are ready");
    var bacon = await baconTask;
    Console.WriteLine("bacon is ready");
    var toast = await toastTask;
    Console.WriteLine("toast is ready");
    Juice oj = PourOJ();
    Console.WriteLine("oj is ready");

    Console.WriteLine("Breakfast is ready!");

    async Task<Toast> makeToastWithButterAndJamAsync(int number)
    {
        var plainToast = await ToastBreadAsync(number);
        ApplyButter(plainToast);
        ApplyJam(plainToast);
        return plainToast;
    }
}
```

Is it not better to be as follows?:

```csharp
static async Task Main(string[] args)
{
    Coffee cup = PourCoffee();
    Console.WriteLine("coffee is ready");
    Task<Egg> eggsTask = FryEggsAsync(2);
    Task<Bacon> baconTask = FryBaconAsync(3);
    var toastTask = makeToastWithButterAndJamAsync(2);

    var eggs = await eggsTask;
    Console.WriteLine("eggs are ready");
    var bacon = await baconTask;
    Console.WriteLine("bacon is ready");
    var toast = await toastTask;
    Console.WriteLine("toast is ready");
    Juice oj = PourOJ();
    Console.WriteLine("oj is ready");

    Console.WriteLine("Breakfast is ready!");

    async Task<Toast> makeToastWithButterAndJamAsync(int number)
    {
        Task<Toast> plainToast = await ToastBreadAsync(number);
        ApplyButter(plainToast);
        ApplyJam(plainToast);
        return plainToast;
    }
}
```

Since you commented above: "The await keyword provides a non-blocking way to start a task, then continue execution when that task completes. "

---
#### Detalles del documento

? *No edite esta sección. Se requiere para docs.microsoft.com ? Vinculación de problema de GitHub.*

* ID: 17d27a0d-c072-8c9f-7e31-3de60bee9ae8
* Version Independent ID: e0c09794-e712-19ff-cb9f-12648d60d4f6
* Content: [Programación asincrónica en C#](https://docs.microsoft.com/es-es/dotnet/csharp/programming-guide/concepts/async/#feedback)
* Content Source: [docs/csharp/programming-guide/concepts/async/index.md](https://github.com/dotnet/docs.es-es/blob/live/docs/csharp/programming-guide/concepts/async/index.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by CeciAc** [comment]
hi team, 
I'm moving this here since the contributor is making a comment about the code and the original documentation
best

**Comment by svick** [comment]
@dimiste Do I understand it correctly that the only thing you changed is to turn `var` into `Task<T>`? Why do you think it's better to specify the type explicitly? Isn't naming the variables `somethingTask` sufficient?

(Also, the third change is incorrect: the value assigned to the `plainToast` variable has already been `await`ed, so it's not a `Task<T>`.)

**Comment by dimiste** [comment]
I thought that in the case of:

`var eggsTask = FryEggsAsync(2); `

the type returned is "Egg", so I modified the variable type to "Task <Egg>", but, I realized that the type returned is really "Task <Egg>". I had not understood logic well.
Therefore, it is not necessary to change the variable type to a specific type.
You are right about the third change, the variable is "Toast" type. I did not have very clear the types returned from the objects, but with your explanation I understood it.
